src/database/mod.rs

```diff
@@ -52,16 +52,22 @@ pub trait DatabaseConnection {
     ) -> Option<bip32::ChildNumber>;
 
     /// Get all UTxOs.
-    fn unspent_coins(&mut self) -> HashMap<bitcoin::OutPoint, Coin>;
+    fn list_unspent_coins(&mut self) -> HashMap<bitcoin::OutPoint, Coin>;
+
+    /// Get all UTxOs.
+    fn list_spending_coins(&mut self) -> HashMap<bitcoin::OutPoint, Coin>;
 
     /// Store new UTxOs. Coins must not already be in database.
     fn new_unspent_coins<'a>(&mut self, coins: &[Coin]);
 
     /// Mark a set of coins as being confirmed at a specified height and block time.
     fn confirm_coins<'a>(&mut self, outpoints: &[(bitcoin::OutPoint, i32, u32)]);
 
-    /// Mark a set of coins as being spent by a specified txid.
+    /// Mark a set of coins as being spent by a specified txid of a pending transaction.
     fn spending_coins<'a>(&mut self, outpoints: &[(bitcoin::OutPoint, bitcoin::Txid)]);
+
+    /// Mark a set of coins as spent by a specified txid at a specified block time.
+    fn spent_coins<'a>(&mut self, outpoints: &[(bitcoin::OutPoint, bitcoin::Txid, u32)]);
```

Comment:
nit: again the method name sounds like you are querying the spent coins. What about having (if you rename the other method i suggested in the previous commit):
- `spend_coins` -> Mark coins as being spent by a specifc txid.
- `confirm_spend` -> Mark the spending transaction of those coins as being confirmed at a given time.